### ISSUE TITLE
[8pt] Revise & enhance Cartridge error handling documentation

### ISSUE THREAD

**Issue by olegrok** [tarantool/cartridge#817]
# Revise & enhance Cartridge error handling documentation

**Product:** Cartridge
**Since:** product version
**Audience/target:** Cartridge app devs
**Root document:**

- https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#error-handling-guidelines
- source: https://github.com/tarantool/cartridge/blob/master/rst/topics/error-handling.rst

**SME:** @rosik, @olegrok 
**Peer reviewer:** @

# Details

In contrast to our default error handling approaches `nil, err` we should directly throw an error to halt triggers. I propose to document it in our error guideline (https://github.com/tarantool/cartridge/blob/master/rst/topics/error-handling.rst).

**Comment by rosik** [comment]
I'd rephrase it as "Revise/rewrite/enhance error handling documentation"
